std/internal/math/biguintcore.d

```diff
@@ -409,6 +409,72 @@ public:
         return data.length * BigDigit.sizeof;
     }
 
+    /**
+        Converts the `BigUint` to a sequence of bytes that represents the
+        native-endian unsigned representation of that number. The sign is
+        completely ignored, so -1,000,000 will produce the same output as
+        1,000,000. The integer will be encoded on the fewest possible bytes.
+        In other words, there will be no leading zeroes.
+
+        Returns: the native-endian unsigned representation of the `BigInt` in
+            the form of a `ubyte[]` array.
+    */
+    ubyte[] toBytes() const pure @system @property nothrow
```

Comment:
Instead of allocating a new buffer would it be feasible to simply return a slice of `this.data`? Is [DIP1000's `scope` qualifier for function return values](https://github.com/dlang/DIPs/blob/master/DIPs/DIP1000.md#scope-function-returns) useful for that?

Reply:
Nope. `toBytes` is 2s complement, while `BigInt` uses signed magnitude.

Reply:
For positive numbers.

Reply:
I suppose that the internal array of `uint`s that is `this.data` could be cast to a `ubyte` array if it is positive, and a slice returned, but I will have to test that. But yeah, when it's negative, that goes out the door, as Biotronic said.